app/apps/server/communication/rest.js

```diff
@@ -260,6 +270,57 @@ export class AppsRestApi {
 			},
 		});
 
+		this.api.addRoute('groupMembers', { authRequired: true }, {
```

Comment:
can you guys please explain why is this endpoint needed? [there are others](https://rocket.chat/docs/developer-guides/rest-api/groups/members/) endpoints that return this information already.

if this is really needed I'd ask to not duplicate the code at least.

Reply:
Compared to the original one, this endpoint bypasses the API rate limitation. Is there any suggestion to refactor it?

Reply:
why is it being rate limited? any reason to call it so often? maybe the UI can cache the result.

Reply:
It's been quite a long time since we needed to introduce this endpoint, right? Maybe it is worth it to test the previous implementation again for us to check if there are any other strategies we could follow (as @sampaiodiego suggested)

Reply:
It's been quite a long time since we needed to introduce this endpoint, right? Maybe it is worth it to test the previous implementation again for us to check if there are any other strategies we could follow (as @sampaiodiego suggested)

Reply:
OK, I will try to implement it with the existing API endpoint.

Reply:
Use cached result when reach the rate limitation now, @sampaiodiego can you please review it https://github.com/RocketChat/Rocket.Chat/pull/15123/commits/8f922b65429ace3a4d4a2879fbf58ce9fce52161?

Reply:
nice.. but there a lot of conflicts 🙈

Reply:
@d-gubert we need to create a new apps-engine beta version that contains both uiKit and externalComponent feature. The `1.12.0-beta.2575` only contains uiKit.